Kernel/Bus/USB/USBHub.cpp

```diff
@@ -71,6 +71,16 @@ ErrorOr<void> Hub::enumerate_and_power_on_hub()
 
     dbgln_if(USB_DEBUG, "USB Hub: Enumerating and powering on for address {}", m_address);
 
+    // Before the hub can be used, it must first be configured via a SET_CONFIGURATION request.
+    // We don't need to set the configuration for the root hub (which has a null m_hub) as we would just ignore that request during root hub emulation anyway.
+    if (m_hub != nullptr) {
+        if (m_configurations.size() < 1)
+            return EINVAL;
+
+        // FIXME: Which configuration should we choose if there is more than one?
```

Comment:
I'd recommend just making an interface on the controller to change the configuration
see https://github.com/SerenityOS/serenity/commit/02120731bbe341c08a111561e27efdfa7bb606db

Reply:
That seems like a general USB subsystem change/improvement, not specific to xHCI support, so I prefer leaving it out of this PR, since it's already quite large.